app/services/certificate_generator.rb

```diff
@@ -61,6 +61,14 @@ def generate_pdf
 
   private
 
+    def certificate_name
+      if @programme.i_belong?
+        @user.school_name
+      else
+        "#{@user.first_name} #{@user.last_name}"
+      end
+    end
```

Comment:
another nitpick - it would be nice if this delegated to a model, perhaps `@programme`.

Reply:
I like this suggestion, thanks :+1: